src/Microsoft.ML.OnnxTransformer/OnnxUtils.cs

```diff
@@ -339,10 +339,7 @@ public static OnnxModel CreateFromBytes(byte[] modelBytes)
         public static OnnxModel CreateFromBytes(byte[] modelBytes, int? gpuDeviceId = null, bool fallbackToCpu = false,
             IDictionary<string, int[]> shapeDictionary = null)
         {
-            var tempModelDir = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
-            Directory.CreateDirectory(tempModelDir);
-
-            var tempModelFile = Path.Combine(tempModelDir, "model.onnx");
+            var tempModelFile = $"{Path.GetTempFileName()}.onnx";
```

Comment:
`Path.GetTempFileName()` will first create the file with that name. So you are creating an empty file every time here and not deleting it.

https://docs.microsoft.com/en-us/dotnet/api/system.io.path.gettempfilename?view=netcore-3.1

> Creates a uniquely named, zero-byte temporary file on disk and returns the full path of that file.

Reply:
Thanks for spotting this. Should be fixed now.